packages/flutter/lib/src/cupertino/page.dart

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+
+const double _kMinFlingVelocity = 1.0;  // screen width per second
+
+// Used for iOS.
+class CupertinoPageTransition extends AnimatedWidget {
+  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
+    begin: FractionalOffset.topRight,
+    end: -FractionalOffset.topRight
+  );
+
+  CupertinoPageTransition({
+    Key key,
+    Animation<double> animation,
+    this.child
+  }) : super(
+    key: key,
+    listenable: _kTween.animate(new CurvedAnimation(
+      parent: animation,
+      curve: new _CupertinoTransitionCurve(null)
+    )
+  ));
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(ianh): tell the transform to be un-transformed for hit testing
+    // but not while being controlled by a gesture.
+    return new SlideTransition(
+      position: listenable,
+      child: new PhysicalModel(
+        shape: BoxShape.rectangle,
+        color: const Color(0x00000000), // we don't need the canvas background, just the elevation
```

Comment:
The color is a required field. 

I looked a bit into the code and tested it on a sample app. Seems like the shadow is always drawn black regardless of the PhysicalModel's color or its child's color in https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/object.dart#L463 which seems like the desired behaviour. 

I think one place where the color here might matter is if somehow the supplied child isn't a rectangle or has holes in it but I think seeing through the model onto the shadow behind it might be a clearer way to signal to the user that he/she needs to deal with the elevated hole than 'filling' it with some arbitrary color.